### ISSUE TITLE
Integration fails to start with new router hub SPNMX42

### ISSUE THREAD

**Issue by DeanK2022** [uvjim/linksys_velop#511]
My ISP just upgraded to Linksys SPNMX42 and now I get this on startup:

```
2024-11-20 16:21:29.084 ERROR (MainThread) [custom_components.linksys_velop] Unexpected error fetching linksys_velop speedtest (55P20M16D01000) data
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/update_coordinator.py", line 382, in _async_refresh
    self.data = await self._async_update_data()
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/linksys_velop/coordinator.py", line 412, in _async_update_data
    result: dict[str, Any] = responses[0][0]
                             ~~~~~~~~~~~~^^^
```

**Comment by DeanK2022** [comment]
I added a debug line at line 412 in coordinator.py:
  _LOGGER.warning(responses)
  
  This is now the output:
  ------------ SNIP -----------------
  2024-11-20 17:23:28.687 WARNING (MainThread) [custom_components.linksys_velop.coordinator] [[], '']
2024-11-20 17:23:28.688 ERROR (MainThread) [custom_components.linksys_velop] Unexpected error fetching linksys_velop speedtest (55P20M16D01000) data
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/update_coordinator.py", line 382, in _async_refresh
    self.data = await self._async_update_data()
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/linksys_velop/coordinator.py", line 413, in _async_update_data
    result: dict[str, Any] = responses[0][0]
                             ~~~~~~~~~~~~^^^
IndexError: list index out of range
  ------------ SNIP -----------------

So basically responses[0][0] is empty

**Comment by uvjim** [comment]
My guess is that as it is new, no Speedtests have been run.  I'll look to guard against that.  As you can imagine, I have a number of Speedtests that have been run so haven't seen this in testing.  I can see in your warning line that there's an empty list and an empty string so I'll look to use those as the guards.

**Comment by uvjim** [comment]
I've just pushed `2024.11.3b0`.  Could you try that and see how you get on please?  The link is: https://github.com/uvjim/linksys_velop/releases/tag/2024.11.3b0 just in case you need it.

**Comment by DeanK2022** [comment]
Wow that was a quick response - thank you and much appreciated!  

I have installed the branch you just made and now I get this:
```
2024-11-21 10:13:18.910 ERROR (MainThread) [custom_components.linksys_velop] Unexpected error fetching linksys_velop speedtest (55P20M16D01000) data
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/update_coordinator.py", line 382, in _async_refresh
    self.data = await self._async_update_data()
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/linksys_velop/coordinator.py", line 426, in _async_update_data
    ret = SpeedtestResults(
          ^^^^^^^^^^^^^^^^^
TypeError: SpeedtestResults.__init__() missing 6 required positional arguments: 'download_bandwidth', 'exit_code', 'latency', 'result_id', 'timestamp', and 'upload_bandwidth'
2024-11-21 10:13:18.911 DEBUG (MainThread) [custom_components.linksys_velop] Finished fetching linksys_velop speedtest (55P20M16D01000) data in 1.704 seconds (success: False)
```

**Comment by uvjim** [comment]
Brilliant.  That's one step further forwards... I know what is happening there.  Just needs another guard.  Fix incoming...

**Comment by uvjim** [comment]
I've just pushed `2024.11.3b1` Could you try that and see how you get on please? The link is: https://github.com/uvjim/linksys_velop/releases/tag/2024.11.3b1 just in case you need it.

Let me know how that one goes.

**Comment by uvjim** [comment]
No problem. Glad it's working.
